File: sms_back_end/database/multi_tenant_school_management/schemas/parent.py
from typing import List
from uuid import UUID
from pydantic import BaseModel

from .parent_student_light import StudentBasic
from .user import UserCreate, UserRead



class ParentBase(BaseModel):
    pass


class ParentCreate(BaseModel):
    user: UserCreate


class ParentRead(BaseModel):
    id: UUID
    user: UserRead

    model_config = {"from_attributes": True}


class ParentWithChildren(ParentRead):
    children: List[StudentBasic] = []

File: sms-frontend/types/parent.ts
import {UserCreate, UserRead} from "@/types/user";
import {UUID} from "node:crypto";

export interface ParentBase {}

export interface ParentCreate {
    user: UserCreate;
}

export interface ParentRead {
    id: UUID;
    user: UserRead;
}

export interface StudentBasic {
    id: UUID;
    admission_number: string;
}

export interface ParentWithChildren extends ParentRead {
    children: StudentBasic[];
}